cli/src/main/kotlin/com/bazel_diff/cli/GetImpactedTargetsCommand.kt

```diff
@@ -66,19 +74,25 @@ class GetImpactedTargetsCommand : Callable<Int> {
 
         validate()
         val deserialiser = DeserialiseHashesInteractor()
-        val from = deserialiser.execute(startingHashesJSONPath, targetType)
-        val to = deserialiser.execute(finalHashesJSONPath, targetType)
+        val from = deserialiser.executeTargetHash(startingHashesJSONPath, targetType)
+        val to = deserialiser.executeTargetHash(finalHashesJSONPath, targetType)
 
-        val impactedTargets = CalculateImpactedTargetsInteractor().execute(from, to)
+        val impactedTargetsStream = if (depsMappingJSONPath != null) {
+            val depsMapping = deserialiser.deserializeDeps(depsMappingJSONPath!!)
+            CalculateImpactedTargetsInteractor().executeWithDistances(from, to, depsMapping).map { (label, metrics) ->
+                "${label}~${metrics.targetDistance}~${metrics.packageDistance}"
+            }.stream()
```

Comment:
I'm open to suggestions on a serialization format here. I picked this since `~` isn't a supported target label character, and doing a full json list-of-dicts could potentially be pretty big with the key names.

Reply:
My only question here would be around if we should prefix the values to make debugging easier, something like 

`label~LABEL~~targetDistance~TARGET_DISTANCE~~packageDistance~PACKAGE_DISTANCE`